src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted.CSharp/BlazorHosted.CSharp.Server/BlazorHosted.CSharp.Server.csproj

```diff
@@ -1,10 +1,16 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework Condition="'$(TargetFrameworkOverride)' == ''">netcoreapp2.1</TargetFramework>
+    <TargetFramework Condition="'$(TargetFrameworkOverride)' != ''">TargetFrameworkOverride</TargetFramework>
   </PropertyGroup>
 
-  <ItemGroup>
+  <ItemGroup Condition="'$(TargetFrameworkOverride)' == ''">
+    <PackageReference Include="Microsoft.AspNetCore.App" Version="2.1.0-preview1-final" />
+    <PackageReference Include="Microsoft.AspNetCore.Blazor.Server" Version="0.2.0-preview1-10152" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFrameworkOverride)' != ''">
     <PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.0" />
```

Comment:
Shouldn't this also be 2.1.0-preview1-final instead of 2.0.0?

Reply:
Yes, but there is now a preview 2 final should I update to that instead?

Reply:
Depends on what the team wants to target, I'll leave that to @danroth27  and @SteveSandersonMS

Also, this PR would have Blazor showing when 2.1 is selected in the dialog, but not 2.0.
I think you would want to add 2.0 as an additional choice in template.json and add the associated conditions in the csproj file.

Note that all the other templates in the aspnet/templating repo are a bit different because they ship with a given CLI, so they only have one framework choice.

Reply:
We'd like the Blazor templates to be available for .NET Core & .NET Framework when using ASP.NET Core 2.0 or ASP.NET Core 2.1 (latest preview)

Reply:
@phenning Thanks for the feedback! Based on your comments, I've implemented a different version of this in #545.